src/opencl_common.c

```diff
@@ -2501,6 +2500,46 @@ void get_compute_capability(int sequential_id, unsigned int *major,
 	clGetDeviceInfo(devices[sequential_id],
 	                CL_DEVICE_COMPUTE_CAPABILITY_MINOR_NV,
 	                sizeof(cl_uint), minor, NULL);
+
+	if (!major) {
+/*
+ * Apple, VCL and some other environments don't expose CL_DEVICE_COMPUTE_CAPABILITY_MINOR_NV
+ * so we need this crap - which is incomplete, best effort matching.
+ * http://en.wikipedia.org/wiki/Comparison_of_Nvidia_graphics_processing_units
+ */
+		char dname[MAX_OCLINFO_STRING_LEN];
+
+		HANDLE_CLERROR(clGetDeviceInfo(devices[sequential_id],
+		                               CL_DEVICE_NAME,
+		                               sizeof(dname), dname, NULL),
+		               "clGetDeviceInfo for CL_DEVICE_NAME");
+
+		// Ampere 8.0
+		if (strstr(dname, "TITAN V") || strstr(dname, "RTX 30") ||
+		    (dname[0] == 'A' && dname[1] >= '1' && dname[1] <= '9'))
+			*major = 8;
```

Comment:
You forget the "RTX A" devices again.

Reply:
I thought I followed your suggestion somewhere but I can't find it now. Are they always called "RTX Axx" or can they be called just "Axx"?

Reply:
As I understand, there are both kinds of naming. For example, there's "RTX A4000" and "RTX A5000", but then there's also "A100", "A10", "A16", "A40" (suffixed by letters indicating form factor).

While we're at it, we could also want to add detection of older Tesla cards... but there are many falling into the many categories we check for below.

Reply:
That whole section is mostly needed for macOS, which is fairly unlikely (I think) to have COFS cards in them. I think we should mostly add stuff to it where actually needed. That said, the bugs you pointed out will be addressed and I can add a few more things.

The canonical solution is to re-add some CUDA code and encourage macOS users (actually all users) to install CUDA. That way we'll get proper numbers (and more).